src/main/java/taxi/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package taxi.service;
+
+import taxi.exception.AuthenticationException;
+import taxi.lib.Inject;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverService driverService;
+
+    @Override
+    public Driver login(String login, String password) throws AuthenticationException {
+        Driver driver = driverService.findByLogin(login);
```

Comment:
what if we don't have driver with such login?
looks like user will not have any message about this

Reply:
i add exception at DriverServiceImpl for this case

Reply:
did you check it?
it will not be visible for user in browser